resources/services/s3/buckets.go

```diff
@@ -451,27 +453,53 @@ func S3Buckets() *schema.Table {
 // ====================================================================================================================
 //                                               Table Resolver Functions
 // ====================================================================================================================
+
 func fetchS3Buckets(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
 	svc := meta.(*client.Client).Services().S3
 	response, err := svc.ListBuckets(ctx, nil)
 	if err != nil {
 		return diag.WrapError(err)
 	}
-	wb := make([]*WrappedBucket, len(response.Buckets))
-	for i, b := range response.Buckets {
-		wb[i] = &WrappedBucket{b, nil, nil}
+
+	var wg sync.WaitGroup
+	buckets := make(chan types.Bucket)
+	errs := make(chan error)
+	for i := 0; i < fetchS3BucketsPoolSize; i++ {
+		wg.Add(1)
+		go fetchS3BucketsWorker(ctx, meta, buckets, errs, res, &wg)
 	}
+	for _, bucket := range response.Buckets {
+		buckets <- bucket
+	}
+	close(buckets)
+	go func() {
+		wg.Wait()
```

Comment:
I would include the `for response.Buckets` (and close buckets chan) in this gofunc as well, otherwise (since we're using unbuffered channels) a deadlock is possible since we're not reading the errors channel (so if we don't have a worker for every bucket the send will fail

I would also check for early cancellation of context and switch those channel writes where to select statements where <-ctx.Done is also checked?